test/test_nestedtensor.py

```diff
@@ -1861,27 +1862,28 @@ def rand_mask(size):
 
         dropout_p = 0.0  # no dropout for reproducibility
 
-        # Success case: no attn_mask set and is_causal=False.
-        actual = torch.nn.functional.scaled_dot_product_attention(
-            query, key, value, attn_mask=None, is_causal=False, dropout_p=dropout_p)
-
-        expected_outputs = []
-        for q, k, v in zip(query.unbind(), key.unbind(), value.unbind()):
-            output = torch.nn.functional.scaled_dot_product_attention(
-                q.unsqueeze(0), k.unsqueeze(0), v.unsqueeze(0), attn_mask=None, dropout_p=dropout_p)
-            expected_outputs.append(output.squeeze(0))
-        expected_output_nested = torch.nested.nested_tensor(expected_outputs)
-        self.assertEqual(actual, expected_output_nested)
-
-        # Error case: explicit attn_mask set.
-        with self.assertRaisesRegex(RuntimeError, "not supported when an explicit attn_mask is set"):
-            torch.nn.functional.scaled_dot_product_attention(
-                query, key, value, attn_mask=attn_mask, dropout_p=dropout_p)
-
-        # Error case: is_causal=True.
-        with self.assertRaisesRegex(RuntimeError, "not supported when is_causal=True"):
-            torch.nn.functional.scaled_dot_product_attention(
-                query, key, value, dropout_p=dropout_p, is_causal=True)
+        with sdp_kernel(enable_flash=False, enable_math=True, enable_mem_efficient=False):
+            # Success case: no attn_mask set and is_causal=False.
+            actual = torch.nn.functional.scaled_dot_product_attention(
+                query, key, value, attn_mask=None, is_causal=False, dropout_p=dropout_p)
+
+            expected_outputs = []
+            for q, k, v in zip(query.unbind(), key.unbind(), value.unbind()):
+                output = torch.nn.functional.scaled_dot_product_attention(
+                    q.unsqueeze(0), k.unsqueeze(0), v.unsqueeze(0), attn_mask=None, dropout_p=dropout_p)
+                expected_outputs.append(output.squeeze(0))
+            expected_output_nested = torch.nested.nested_tensor(expected_outputs)
+            self.assertEqual(actual, expected_output_nested)
+
+            # Error case: explicit attn_mask set.
+            with self.assertRaisesRegex(RuntimeError, "not supported when an explicit attn_mask is set"):
+                torch.nn.functional.scaled_dot_product_attention(
+                    query, key, value, attn_mask=attn_mask, dropout_p=dropout_p)
+
+            # Error case: is_causal=True.
+            with self.assertRaisesRegex(RuntimeError, "not supported when is_causal=True"):
```

Comment:
I want to check this, cause I don't know if this fails with the fused kernels

Reply:
I think I shouldn't have indented this code block with a `sdp_kernel(enable_flash=False, enable_math=True, enable_mem_efficient=False)`, both num_head and seq_len are ragged here so it should always dispatch to math, but I should still be checking that it doesn't dispatch to the wrong kernel, my bad